pkgs/applications/video/davinci-resolve/default.nix

```diff
@@ -0,0 +1,179 @@
+{ stdenv
+, lib
+, cacert
+, curl
+, runCommandLocal
+, targetPlatform
+, unzip
+, appimage-run
+, addOpenGLRunpath
+, libGLU
+, xorg
+, buildFHSUserEnv
+, bash
+, writeText
+, ocl-icd
+, xkeyboard_config
+, glib
+, libarchive
+}:
+let
+  davinci = (
+    stdenv.mkDerivation rec {
+      pname = "davinci-resolve";
+      version = "17.4.3";
+
+      nativeBuildInputs = [ unzip appimage-run addOpenGLRunpath ];
+
+      # Pretty sure, there are missing dependencies ...
+      buildInputs = [ libGLU xorg.libXxf86vm ];
+
+      src = runCommandLocal "${pname}-src.zip"
+        rec {
+          outputHashMode = "recursive";
+          outputHashAlgo = "sha256";
+          outputHash = "0hq374n26mbcds8f1z644cvnh4h2rjdrbxxxbj4p34mx9b04ab28";
+
+          impureEnvVars = lib.fetchers.proxyImpureEnvVars;
+
+          nativeBuildInputs = [ curl ];
+
+          # ENV VARS
+          SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+
+          DOWNLOADID = "5efad1a052e8471989f662338d5247f1";
+          REFERID = "263d62f31cbb49e0868005059abcb0c9";
+          SITEURL = "https://www.blackmagicdesign.com/api/register/us/download/${DOWNLOADID}";
+
+          USERAGENT = builtins.concatStringsSep " " [
+            "User-Agent: Mozilla/5.0 (X11; Linux ${targetPlatform.linuxArch})"
+            "AppleWebKit/537.36 (KHTML, like Gecko)"
+            "Chrome/77.0.3865.75"
+            "Safari/537.36"
+          ];
+
+          REQJSON = ''
```

Comment:
This is a completely unnecessary suggestion, but maybe instead use:
```nix
REQJSON = builtins.toJSON {
  "firstname" = "NixOS";
  "lastname" = "Linux";
  "email" = "someone@nixos.org";
  "phone" = "202-555-0194";
  "country" = "us";
  "state" = "New York";
  "city" = "AUR";
  "product" = "DaVinci Resolve";
};
```

Reply:
nice

Reply:
Another unnecessary suggestion: `"city" = "Nixpkgs"` instead of `"city" = "AUR"` :)

Reply:
i also thought about it. on the website they have a drop down. we might want to choose a real location from the list... `AUR` sounds more like a real city, `Nixpkgs` is obviously fake data. also i think one can say we come from the same city als Arch... the linux city ;)

Reply:
I created a forum post for this topic, so we can decide once and for all.

https://discourse.nixos.org/t/fake-user-data-for-download/16875